deploy-manage/upgrade/deployment-or-cluster/upgrade-717.md

```diff
@@ -1,7 +1,501 @@
+- {{es}}
+- {{kib}}
+- Integrations Server ({{fleet-server}} and APM)
+- Enterprise Search
+::::
+
+
+::::{applies-item} eck:
+In ECK, upgrades are performed declaratively by updating the `spec.version` field in the resource manifest. Once the new version is applied, the operator automatically orchestrates the rolling upgrade, ensuring that each component is upgraded safely and in the correct order.
+
+To upgrade your cluster to 8.19, follow the steps in [Upgrade on ECK](/deploy-manage/upgrade/deployment-or-cluster/upgrade-on-eck.md), and start upgrading the {{es}} and {{kib}} resources that represent the cluster.
+
+:::{note}
+For more information on how ECK manages upgrades and how to tune its behavior, refer to [Nodes orchestration](/deploy-manage/deploy/cloud-on-k8s/nodes-orchestration.md).
+:::
+
+After upgrading {{es}} and {{kib}}, upgrade any [other Elastic applications](/deploy-manage/deploy/cloud-on-k8s/orchestrate-other-elastic-applications.md) connected to the cluster, such as {{fleet-server}} or Elastic APM.
+::::
+
+::::{applies-item} self:
+To upgrade your cluster to 8.19, follow the steps in [Upgrade self-managed {{stack}}](https://www.elastic.co/guide/en/elastic-stack/8.19/upgrading-elastic-stack-on-prem.html).
+
+Make sure to upgrade all components in the specified order.
+::::
+
+:::::
+
+### 8.19 upgrade validation [819-validation]
+
+:::{include} _snippets/upgrade-validation.md
+:::
+
+### Upgrade ingest components and {{es}} client libraries to 8.19.x
+
+Before upgrading to {{version.stack}}, ensure that all ingest components and {{es}} clients are upgraded to version 8.19.x, as 7.x versions aren’t compatible with {{stack}} 9.x, according to the [Product compatibility support matrix](https://www.elastic.co/support/matrix#matrix_compatibility).
+
+For more details, refer to the documentation of the following products and client libraries:
+
+::::{dropdown} 8.19 {{stack}} ingest components
+* {{beats}}: [Upgrade instructions](https://www.elastic.co/guide/en/beats/libbeat/8.19/upgrading.html)
+* {{ls}}: [Upgrade instructions](https://www.elastic.co/guide/en/logstash/8.19/upgrading-logstash.html)
+* {{fleet}} managed {{agent}}: [Upgrade instructions](https://www.elastic.co/guide/en/fleet/8.19/upgrade-elastic-agent.html)
+* Standalone {{agent}}: [Upgrade instructions](https://www.elastic.co/guide/en/fleet/8.19/upgrade-standalone.html)
+* Elastic APM: [Upgrade instructions](https://www.elastic.co/guide/en/observability/8.19/apm-upgrade.html).
+* Enterprise Search: [Upgrade and migration guides](https://www.elastic.co/guide/en/enterprise-search/8.19/upgrading-and-migrating.html)
+
+:::{note}
+The Elastic APM Server and Enterprise Search components require manual upgrade only in ECK-managed or self-managed environments. In {{ech}} and {{ece}}, these components are upgraded automatically as part of the deployment upgrade process.
+:::
+
+::::
+
+::::{dropdown} 8.19 {{es}} client libraries
```

Comment:
eland missing?

Reply:
looks like [it was unversioned ](https://docs.google.com/spreadsheets/d/1bQQVc-QwwXDSd24Vw6Q62dnn_9bxsBeAzeWJAdyxdqI/edit?gid=1897179853#gid=1897179853)so only the one set of docs exists. not sure if there's a migration impact